### PR TITLE
resources loop

### PR DESCRIPTION
**Pull request by odra**
## Verification

* Build the image
* Run the backup job with the custom image
* Check if all objects were properly backed up

### CODE REVIEW

**Comment by matskiv** [comment]
Reviewing and testing now.

File: image/tools/lib/component/resources.sh [hunk 1]
```diff
@@ -39,10 +39,11 @@ function backup_resource {
             echo '---' > /tmp/${type}.yaml
             for obj in $(oc get ${type} -n ${ns} | tr -s ' ' | cut -d ' ' -f 1 |  tail -n +2); do
                 echo '-' >> /tmp/${type}.yaml
```
Comment (matskiv) [inline image/tools/lib/component/resources.sh]: This line is no longer required.

**Comment by matskiv** [comment]
I looked at restoration SOP, and it looks like once all .gz archives are unpacked, we are expecting "$NS-\*" files.
Current implementation creates enmasse-messagingusers.yaml.gz with this structure "tmp/messagingusers.\*.yaml".

We should update SOP or update filenames and archive them without tmp/ folder.

**Comment by odra** [comment]
Yeah the sop needs an update as well but I could add the ns var into the file name as well

**Comment by odra** [comment]
@matskiv sent a tiny update to remove the /tmp from the file path

File: image/tools/lib/component/resources.sh [hunk 2]
```diff
@@ -39,10 +39,11 @@ function backup_resource {
...
-                echo "$(oc get ${type}/${obj} -n ${ns} -o yaml --export | sed 's/^/  /')" >> /tmp/${type}.yaml
-                cat /tmp/${type}.yaml | gzip > ${dest}/archives/${ns}-${type}.yaml.gz
-                rm -f /tmp/${type}.yaml
+                echo "$(oc get ${type}/${obj} -n ${ns} -o yaml --export)" > /tmp/${type}.${obj}.yaml
```
Comment (matskiv) [inline image/tools/lib/component/resources.sh]: if we change output filename, to start with `${ns}-`, we won't need SOP update IMHO

**Comment by maleck13** [comment]
Once merged lets get this into 1.4 as well

**Comment by odra** [comment]
still needs a final verification

File: image/tools/lib/component/resources.sh [hunk 3]
```diff
@@ -39,10 +39,11 @@ function backup_resource {
             echo '---' > /tmp/${type}.yaml
             for obj in $(oc get ${type} -n ${ns} | tr -s ' ' | cut -d ' ' -f 1 |  tail -n +2); do
                 echo '-' >> /tmp/${type}.yaml
-                echo "$(oc get ${type}/${obj} -n ${ns} -o yaml --export | sed 's/^/  /')" >> /tmp/${type}.yaml
-                cat /tmp/${type}.yaml | gzip > ${dest}/archives/${ns}-${type}.yaml.gz
-                rm -f /tmp/${type}.yaml
+                echo "$(oc get ${type}/${obj} -n ${ns} -o yaml --export)" > /tmp/${ns}-${type}.${obj}.yaml
             done
+            (cd /tmp; find -name "*.yaml" | tar -cvzf  "${dest}/archives/${ns}-${type}.yaml.gz" -T -)
```
Comment (matskiv) [inline image/tools/lib/component/resources.sh]: This produces .tar.gz archive, and if one follows SOP and uses gunzip - it won't unpack this archive.

File: image/tools/lib/component/resources.sh (hunk 1 above)
Comment (matskiv) [reply image/tools/lib/component/resources.sh]: and `echo '---' > /tmp/${type}.yaml` also.
/tmp/${type}.yaml doesn't have any real content besides dashes.

File: image/tools/lib/component/resources.sh (hunk 3 above)
Comment (odra) [reply image/tools/lib/component/resources.sh]: ok but all other components are using tar we should update the sop then

**Review by matskiv — APPROVED** [review]
Looks good.

**Comment by maleck13** [comment]
@odra ok to merge, release and cherry pick

**Pull request merged**